tests/testthat/test-lazyframe-frame.R

```diff
@@ -797,6 +797,26 @@ test_that("fill_null(): basic usage", {
   )
 })
 
+test_that("fill_null() fills categoricals if fill is character", {
+  df <- pl$DataFrame(
+    num = c(1, 2, NA),
+    str = c("a", "b", NA),
+    cat_lex = c("a", "b", NA),
+    cat_phy = c("a", "b", NA)
+  )$cast(cat_lex = pl$Categorical("lexical"), cat_phy = pl$Categorical("physical"))
+
+  expect_query_equal(
+    .input$fill_null("foobar"),
+    df,
+    pl$DataFrame(
+      num = c(1, 2, NA),
+      str = c("a", "b", "foobar"),
+      cat_lex = c("a", "b", "foobar"),
+      cat_phy = c("a", "b", "foobar")
+    )$cast(cat_lex = pl$Categorical("lexical"), cat_phy = pl$Categorical("physical"))
+  )
```

Comment:
This throws a message about how categorical remapping is expensive (also in py-polars), but it doesn't fail the tests.